rust/processor/src/processors/econia_processor.rs

```diff
@@ -0,0 +1,212 @@
+use super::processor_trait::{ProcessingResult, ProcessorTrait};
+use crate::{
+    models::default_models::transactions::{TransactionDetail, TransactionModel},
+    utils::database::{execute_with_better_error, PgDbPool},
+};
+
+use aptos_indexer_protos::transaction::v1::Transaction;
+use async_trait::async_trait;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Utc};
+use diesel::{result::Error, PgConnection};
+use econia_db::models::MarketRegistrationEvent;
+use econia_db::schema::market_registration_events;
+use serde_json::Value;
+use std::str::FromStr;
+use std::{collections::HashMap, fmt::Debug};
+
+pub const NAME: &str = "econia_processor";
+pub struct EconiaTransactionProcessor {
+    connection_pool: PgDbPool,
+    econia_address: String,
+}
+
+impl EconiaTransactionProcessor {
+    pub fn new(connection_pool: PgDbPool, econia_address: String) -> Self {
+        Self {
+            connection_pool,
+            econia_address,
+        }
+    }
+}
+
+impl Debug for EconiaTransactionProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = &self.connection_pool.state();
+        write!(
+            f,
+            "DefaultTransactionProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+fn hex_to_string(hex: &str) -> String {
+    let hex_no_prefix = &hex[2..];
+    let hex_bytes = hex::decode(hex_no_prefix).unwrap();
+    String::from_utf8(hex_bytes).unwrap()
+}
+
+fn opt_value_to_big_decimal(value: Option<&Value>) -> BigDecimal {
+    let string = value
+        .and_then(Value::as_str)
+        .expect("key not found or not a string");
+    BigDecimal::from_str(string).expect("Failed to parse BigDecimal")
+}
```

Comment:
Here, I would either not pass an option, or pass and return an option. `except` acts just like `unwrap` but with a message. So I would not use it in this function.

```suggestion
fn opt_value_to_big_decimal(value: &Value) -> anyhow::Result<BigDecimal> {
    let string = value
        .and_then(Value::as_str)?;
    BigDecimal::from_str(string)?
}
```

Note: here, `Value::and_then` and `BigDecimal::from_str` do not return the same error. This can make error handling in your function cumbersome. To fix this, there is a crate called `anyhow` (already used in the project) that introduces a new `Result<T>` type, that basically can accept any error. `?` propagates the error, so basically it means "if this is an error, return `Err(the error)`, else, just unwrap".

Reply:
Ended up doing this I believe.